rma/wizard/rma_delivery_views.xml

```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2020 Tecnativa - Ernesto Tejeda
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="rma_redelivery_wizard_view_form" model="ir.ui.view">
+        <field name="name">rma.delivery.wizard.form</field>
+        <field name="model">rma.delivery.wizard</field>
+        <field name="arch" type="xml">
+            <form>
+                <group>
+                    <group>
+                        <field name="scheduled_date" />
+                        <field
+                            name="warehouse_id"
+                            attrs="{'invisible': [('type', '!=', 'replace')]}"
+                        />
+                        <field
+                            name="rma_return_grouping"
+                            attrs="{'invisible': ['|', ('type', '=', 'replace'), ('rma_count', '=', 1)]}"
+                        />
+                    </group>
+                    <group>
+                        <field name="uom_category_id" invisible="1" />
+                        <field
+                            name="product_id"
+                            attrs="{'invisible': ['|', ('type', '!=', 'replace'), ('rma_count', '>', 1)], 'required': [('type', '=', 'replace'), ('rma_count', '=', 1)]}"
+                        />
+                        <label
+                            for="product_uom_qty"
+                            attrs="{'invisible': [('rma_count', '>', 1)]}"
+                        />
+                        <div
+                            class="o_row"
+                            attrs="{'invisible': [('rma_count', '>', 1)]}"
+                        >
+                            <field
+                                name="product_uom_qty"
+                                attrs="{'required': [('rma_count', '=', 1)]}"
+                            />
+                            <field
+                                name="product_uom"
+                                groups="uom.group_uom"
+                                attrs="{'required': [('rma_count', '=', 1)]}"
+                                domain="[('category_id', '=', uom_category_id)]"
+                            />
```

Comment:
Another error that I'd say is related to the other one and the way the groups are handled for the final view rendering and the attributes we get.

```suggestion
                            <field
                                name="product_uom"
                                groups="uom.group_uom"
                                attrs="{'required': [('rma_count', '=', 1)]}"
                                domain="[('category_id', '=', uom_category_id)]"
                            />
                            <field name="product_uom" groups="!uom.group_uom" invisible="1"/>
```

The error with no uom group:

```
  File "/mnt/data/odoo-addons-dir/rma/wizard/rma_delivery.py", line 82, in action_deliver
    rma.create_replace(
  File "/mnt/data/odoo-addons-dir/rma/models/rma.py", line 1218, in create_replace
    self._action_launch_stock_rule(scheduled_date, warehouse, product, qty, uom)
  File "/mnt/data/odoo-addons-dir/rma/models/rma.py", line 1303, in _action_launch_stock_rule
    self.env["procurement.group"].run([procurement])
  File "/opt/odoo/addons/stock/models/stock_rule.py", line 446, in run
    float_is_zero(procurement.product_qty, precision_rounding=procurement.product_uom.rounding)
  File "/opt/odoo/odoo/tools/float_utils.py", line 123, in float_is_zero
    epsilon = _float_check_precision(precision_digits=precision_digits,
  File "/opt/odoo/odoo/tools/float_utils.py", line 29, in _float_check_precision
    assert precision_rounding is None or precision_rounding > 0,\
AssertionError: precision_rounding must be positive, got 0.0
```

Reply:
Yes, that one is the key: to add the fields protected with groups as invisible for the framework to prefetch them anyway and be able to operate with them in defaults, compute, etc. Rechecked now all the affected ones. Anyway, we leave the defensive code about the existence of the dictionary key.